runtime/v2/runc/container.go

```diff
@@ -30,7 +30,7 @@ import (
 	"github.com/containerd/cgroups/v3/cgroup1"
 	cgroupsv2 "github.com/containerd/cgroups/v3/cgroup2"
 	"github.com/containerd/console"
-	"github.com/containerd/containerd/api/runtime/task/v2"
```

Comment:
Maybe we need to verify the binary size after introduce grpc for shim.
As far as I know, we should keep it small as we could

Reply:
runc shim still uses TTRPC.
We allow GRPC based shim as an option, but we don't enforce it and still expect TTRPC by default.

Reply:
Never mind. The shim already depends on grpc. So this pr won't change this fact.

```bash
➜  containerd git:(release/1.5) go version -m bin/containerd-shim-runc-v2

bin/containerd-shim-runc-v2: go1.20.10
        path    github.com/containerd/containerd/cmd/containerd-shim-runc-v2
        mod     github.com/containerd/containerd        (devel)
        dep     github.com/cilium/ebpf  v0.6.2  h1:iHsfF/t4aW4heW2YKfeHrVPGdtYTL4C4KocpM8KTSnI=
        dep     github.com/containerd/cgroups   v1.0.3  h1:ADZftAkglvCiD44c77s5YmMqaP2pzVCFZvBmAlBdAP4=
        dep     github.com/containerd/console   v1.0.2  h1:Pi6D+aZXM+oUw1czuKgH5IJ+y0jhYcwBJfx5/Ghn9dE=
        dep     github.com/containerd/fifo      v1.0.0  h1:6PirWBr9/L7GDamKr+XM0IeUFXu5mf3M/BPpH9gaLBU=
        dep     github.com/containerd/go-runc   v1.0.0  h1:oU+lLv1ULm5taqgV/CJivypVODI4SUz1znWjv3nNYS0=
        dep     github.com/containerd/ttrpc     v1.1.0  h1:GbtyLRxb0gOLR0TYQWt3O6B0NvT8tMdorEHqIQo/lWI=
        dep     github.com/containerd/typeurl   v1.0.2  h1:Chlt8zIieDbzQFzXzAeBEF92KhExuE4p9p92/QmY7aY=
        dep     github.com/coreos/go-systemd/v22        v22.3.2 h1:D9/bQk5vlXQFZ6Kwuu6zaiXJ9oTPe68++AzAJc1DzSI=
        dep     github.com/docker/go-units      v0.4.0  h1:3uh0PgVws3nIA0Q+MwDC8yjEPf9zjRfZZWXZYDct3Tw=
        dep     github.com/godbus/dbus/v5       v5.0.4  h1:9349emZab16e7zQvpmsbtjc18ykshndd8y2PG3sgJbA=
        dep     github.com/gogo/protobuf        v1.3.2  h1:Ov1cvc58UF3b5XjBnZv7+opcTcQFZebYjWzi34vdm4Q=
        dep     github.com/golang/protobuf      v1.5.0
        =>      github.com/golang/protobuf      v1.3.5  h1:F768QJ1E9tib+q5Sc8MkdJi1RxLTbRcTf8LJV56aRls=

        dep     github.com/hashicorp/errwrap    v1.0.0  h1:hLrqtEDnRye3+sgx6z4qVLNuviH3MR5aQ0ykNJa/UYA=
        dep     github.com/hashicorp/go-multierror      v1.0.0  h1:iVjPR7a6H0tWELX5NxNe7bYopibicUzc7uPribsnS6o=
        dep     github.com/moby/sys/mountinfo   v0.4.1  h1:1O+1cHA1aujwEwwVMa2Xm2l+gIpUHyd3+D+d7LZh1kM=
        dep     github.com/opencontainers/go-digest     v1.0.0  h1:apOUWs51W5PlhuyGyz9FCeeBIOUDA/6nW8Oi/yOhh5U=
        dep     github.com/opencontainers/runtime-spec  v1.0.3-0.20210326190908-1c3f411f0417    h1:3snG66yBm59tKhhSPQrQ/0bCrv1LQbKt40LnUPiUxdc=
        dep     github.com/pkg/errors   v0.9.1  h1:FEBLx1zS214owpjy7qsBeixbURkuhQAwrK5UwLGTwt4=
        dep     github.com/sirupsen/logrus      v1.8.1  h1:dJKuHgqk1NNQlqoA6BTlM1Wf9DOH3NBjQyu0h9+AZZE=
        dep     golang.org/x/sys        v0.0.0-20220722155257-8c9f86f7a55f      h1:v4INt8xihDGvnrfjMDVXGxw9wrfxYyCjk0KbXjhR55s=
        dep     google.golang.org/genproto      v0.0.0-20201110150050-8816d57aaa9a
        =>      google.golang.org/genproto      v0.0.0-20200224152610-e50cd9704f63      h1:YzfoEYWbODU5Fbt37+h7X16BWQbad7Q4S6gclTKFXM8=

        dep     google.golang.org/grpc  v1.33.2
        =>      google.golang.org/grpc  v1.27.1 h1:zvIju4sqAGvwKspUQOhwnpcqSbzi7/H6QomNNjTL4sk=

        build   -buildmode=exe
        build   -compiler=gc
        build   -ldflags="-X github.com/containerd/containerd/version.Version=v1.5.18 -X github.com/containerd/containerd/version.Revision=39bb06f98f17c7a226b10269d325c861585b2389 -X github.com/containerd/containerd/version.Package=github.com/containerd/containerd -extldflags \"-static\" -s -w"
        build   -tags=no_btrfs
        build   CGO_ENABLED=0
        build   GOARCH=amd64
        build   GOOS=linux
        build   GOAMD64=v1
        build   vcs=git
        build   vcs.revision=39bb06f98f17c7a226b10269d325c861585b2389
        build   vcs.time=2023-02-15T22:40:52Z
        build   vcs.modified=false

```

The each release's binary size is here (based on golang:1.20.10)

| Release | Binary | Size |
| -- | -- | -- |
| 1.2 | containerd-shim(v1)          | 5.0M |
| 1.2 | containerd-shim-runc-v1 | 6.0M|
| 1.3 | containerd-shim-runc-v2 | 6.9M |
| 1.4 | containerd-shim-runc-v2 | 7.8M |
| 1.5 | containerd-shim-runc-v2 | 7.9M |
| 1.6 | containerd-shim-runc-v2 | 9.3M |
| 1.7 | containerd-shim-runc-v2 | 12M  |
| this pr | containerd-shim-runc-v2 | 12M |


I am thinking why we keep the shim size small. 
Since the shim can't be restart, even if containerd has upgraded, there are server versions of running shim in the host. The page-cache can be reused. 
Since the v1.7 has been increased into 12MiB, we can consider to make it small in the following releases.

Reply:
I'll take a look in a separate PR.